src/components/Modals/ModalAddPost/ModalAddPost.tsx

```diff
@@ -1,49 +1,69 @@
-import './ModalAddPost.css';
 import { Modal } from '../../Modal/Modal';
-import { users } from '../../../utils/tempDB';
 import { useState } from 'react';
 import { openModal } from '../../../redux/modalSlice';
-import { useDispatch } from 'react-redux';
+import { useAppDispatch, useAppSelector } from '../../../redux/hooks';
+import Form from '../../Form/Form';
+import { formatImgUrl } from '../../../utils/helpers';
+import { getUser, getModal } from '../../../redux/selectors';
+import { ResidentData } from '../../../types/resident';
+import { useImageUrl } from '../../../hooks/useImageUrl';
 
 type AddPostFormProps = {
   formName: string;
-  userId: number;
   onClose: () => void;
   onNext: (
-    resident: { id: number; name: string; avatarUrl: string; species: string },
+    resident: { _id: string; name: string; avatar: string; species: string },
     photoUrl: string
   ) => void;
+  chosenResident?: ResidentData;
 };
 
-function ModalAddPost({ formName, onClose, userId, onNext }: AddPostFormProps) {
-  const dispatch = useDispatch();
-  const [selectedResident, setSelectedResident] = useState({
-    id: 0,
-    name: 'Choose your resident',
-    avatarUrl: '',
-    species: '',
-  });
+function ModalAddPost({
+  formName,
+  onClose,
+  onNext,
+
+}: AddPostFormProps) {
+  const dispatch = useAppDispatch();
+  const user = useAppSelector(getUser);
+  const modal = useAppSelector(getModal)
+  const residents = user?.residents;
+  const [selectedResident, setSelectedResident] = useState(
+    modal?.chosenResident || {
+      _id: '',
+      name: 'Choose your resident',
+      avatar: '',
+      species: '',
+    }
+  );
   const [photoUrl, setPhotoUrl] = useState('');
   const [optionsVisibility, setOptionsVisibility] = useState(false);
+  const [urlError, setUrlError] = useState('');
 
   const toggleOptionsVisibility = () =>
     setOptionsVisibility(!optionsVisibility);
 
   const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
-    onNext(selectedResident, photoUrl);
+    const cleanUrl = formatImgUrl(photoUrl, setUrlError);
+
+    if (!cleanUrl) {
+      console.error('Invalid URL, skipping plant suggestion.');
+      return;
+    }
+
+    onNext(selectedResident, cleanUrl);
   };
 
   return (
     <>
```

Comment:
`<>` empty tags are not needed if you have a common tag that wraps the whole markup of the component